### PR TITLE
Feature/333 settings

### PR DESCRIPTION
**Pull request by pierreTklein**
# Description

Settings schema for opening registration, closing registration, confirming your attendance.
Setting schema:
```
{
    openTime: {
        type: Date,
        default: 0
    },
    closeTime: {
        type: Date,
        default: Date.now() + 31104000000 // Add a year from now.
    },
    confirmTime: {
        type: Date,
        default: Date.now() + 31104000000 + 2628000000 // 1 year and 1 month from now.
    }
}
```
Fixes #344 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Successful Patch
- [x] Successful Get

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by pierreTklein** [comment]
This is a WIP, will lyk when it's done

**Comment by pierreTklein** [comment]
@YiFeiZhang2 should be done.

**Comment by pierreTklein** [comment]
@krubenok your settings request 👍

File: services/settings.service.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    if (existingSetting) {
+        return Settings.findOneAndUpdate({}, settingsDetails, logger.queryCallbackFactory(TAG, "settings", {}))
+    } else {
+        const setting = new Settings(settingsDetails);
+        return setting.save();
+    }
+}
+
+/**
+ * @function getRole
```
Comment (YiFeiZhang2) [inline services/settings.service.js]: Typo - should be 'getSettings'

Comment (YiFeiZhang2) [reply services/settings.service.js]: Nitpick, but you might as well put ```@return {Promise<Settings>} The promise will resolve to a Settings object if retrieval was successful.``` or something.

**Pull request merged**